test/all.spec.ts

```diff
@@ -270,3 +276,213 @@ describe('validation', () => {
 		expect(f).to.not.throw();
 	});
 });
+
+describe('environment files', () => {
+	const { Readable } = require('stream');
+
+	const createEnvVarFileContent = (
+		envVars: Array<{ var: string; val: string }>,
+	) =>
+		envVars.reduce((accu, entry, idx) => {
+			return idx === 0
+				? `${entry.var}=${entry.val}\n`
+				: accu + `${entry.var}=${entry.val}\n`;
+		}, '');
+
+	const createStreamCallback = (
+		envVars: Array<{ var: string; val: string }>,
+	) => {
+		return (filePath) => {
+			const envString = envVars.reduce((accu, entry, idx) => {
+				return idx === 0
+					? `${entry.var}=${entry.val}`
+					: accu + '\n' + `${entry.var}=${entry.val}`;
+			}, '');
+			if (filePath) {
+				return Readable.from(envString);
+			} else {
+				return undefined;
+			}
+		};
+	};
+
+	const testCallback = async (filePath: string): Promise<fs.ReadStream> => {
+		return new Promise((resolve, reject) => {
+			fs.realpath(filePath, (err, resolvedPath) => {
+				if (err) {
+					reject(err);
+				}
+				resolve(resolvedPath);
+			});
+		})
+			.then((canonicalPath) => {
+				if (!(canonicalPath as string).startsWith(path.resolve(__dirname))) {
+					throw new ValidationError(
+						`Canonical path outsite project folder (${__dirname}) not allowed: ${canonicalPath}`,
+					);
+				}
+				return fs.createReadStream(canonicalPath as string);
+			})
+			.catch((err) => {
+				throw new Error(`testCallback error : ${err}`);
+			});
+	};
+
+	it('environment from file', async () => {
```

Comment:
nit:  seems like you need a bit more description for the test, like maybe `should read environment from file`

Reply:
Thanks, added better test description